lib/internal/modules/esm/loader.js

```diff
@@ -159,18 +159,16 @@ function nextHookFactory(chain, meta, { validateArgs, validateOutput }) {
       // Set when next<HookName> is actually called, not just generated.
       if (generatedHookIndex === 0) { meta.chainFinished = true; }
 
-      // `context` is an optional argument that only needs to be passed when changed
-      switch (args.length) {
-        case 1: // It was omitted, so supply the cached value
-          ArrayPrototypePush(args, meta.context);
-          break;
-        case 2: // Overrides were supplied, so update cached value
-          ObjectAssign(meta.context, args[1]);
-          break;
+      const argsToApply = [args[0], args[1]];
```

Comment:
This code is relying on `Array.prototype[1]` and `Object.prototype[1]` being undefined, can we avoid that?
```suggestion
      const argsToApply = ArrayPrototypeSlice(args, 0, 2);
```

Reply:
I think my named args suggestion also avoids this problem.
https://github.com/nodejs/node/pull/44109#discussion_r936039541

Reply:
Ah, so it does (both). Will switch to the named vars tomorrow.